server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -95,6 +95,8 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion NORI_DUPLICATES = def(8_500_007, Version.LUCENE_9_9_0);
     public static final IndexVersion UPGRADE_LUCENE_9_9_1 = def(8_500_008, Version.LUCENE_9_9_1);
     public static final IndexVersion ES_VERSION_8_12_1 = def(8_500_009, Version.LUCENE_9_9_1);
+    public static final IndexVersion NEW_INDEXVERSION_FORMAT = def(8_501_00_0, Version.LUCENE_9_9_1);
+    public static final IndexVersion UPGRADE_LUCENE_9_9_2 = def(8_502_00_0, Version.LUCENE_9_9_2);
```

Comment:
@thecoop Now that we've got the _main_ changes sorted out, I wanna check how to proceed here with the 8.12.1 backport. I just aligned the versions. `NEW_INDEXVERSION_FORMAT` is not used at all, just backported it for alignment with _main_.

Reply:
8.12.1 shouldn't know about any of the index versions beyond 8500010 for lucene 9.9.2 - and including them here will make 8.12 use those versions, which we don't want.

Reply:
removed the superfluous versions and added 8500010.   Should we include this, 8500010, in _main_ too for consistency ?